diff --git a/wayf_dk_login.admin.inc b/wayf_dk_login.admin.inc
index bb35733..78f969e 100644
--- a/wayf_dk_login.admin.inc
+++ b/wayf_dk_login.admin.inc
@@ -11,7 +11,6 @@
  * @see system_settings_form()
  */
 function wayf_dk_login__settings_form() {
-
   $idp_metadata_default = wayf_dk_login__get_ipd_metadata('test');
 
   $form['tabs'] = array(
@@ -339,7 +338,38 @@ function wayf_dk_login__settings_form() {
 
   $form['#attached']['css'] = array(drupal_get_path('module', 'wayf_dk_login') . '/admin.css');
 
-  return system_settings_form($form);
+  $form = system_settings_form($form);
+
+  // Add extra submit to enable scoping.
+  $form['#submit'][] = 'wayf_dk_login__settings_form_submit';
+
+  return $form;
+}
+
+/**
+ * Submit handler to the settings form.
+ *
+ * Enable scoping based on allowed organizations.
+ */
+function wayf_dk_login__settings_form_submit($form, $form_state) {
+  $scopes = array();
+  // Get all identity providers.
+  $idps = variable_get('wayf_dk_login_organizations_list', FALSE);
+  if ($idps) {
+    // Lock at allowed organizations to get scopes.
+    $orgs = array_filter($form_state['values']['wayf_dk_login_organizations_active']);
+    foreach ($orgs as $active) {
+      foreach ($idps as $endpoint => $idp) {
+        if ($idp['schacHomeOrganization'] == $active) {
+          // Scope endpoint found.
+          $scopes[] = $endpoint;
+        }
+      }
+    }
+  }
+
+  // Save scopes.
+  variable_set('wayf_dk_login_scope_idplist', $scopes);
 }
 
 /**
diff --git a/wayf_dk_login.install b/wayf_dk_login.install
index 12139e5..8018459 100644
--- a/wayf_dk_login.install
+++ b/wayf_dk_login.install
@@ -43,6 +43,7 @@ function wayf_dk_login_uninstall() {
     'wayf_dk_login_user_localscope',
     'wayf_dk_login_contact_name',
     'wayf_dk_login_contact_mail',
+    'wayf_dk_login_scope_idplist',
   );
 
   foreach ($variables as $variable) {
diff --git a/wayf_dk_login.module b/wayf_dk_login.module
index 785aace..fd1ef2e 100644
--- a/wayf_dk_login.module
+++ b/wayf_dk_login.module
@@ -116,8 +116,11 @@ function wayf_dk_login__endpoint() {
       module_invoke_all('wayf_dk_login_pre_authenticate');
     }
 
+    // Get scopes based on allowed organizations.
+    $scopes = variable_get('wayf_dk_login_scope_idplist', array());
+
     $sporto = new WAYF\SPorto($sporto_config);
-    $result = $sporto->authenticate();
+    $result = $sporto->authenticate($scopes);
 
     // Give other modules an change to alter the auth result after redirects.
     drupal_alter('wayf_dk_login_authenticate_result', $result);
